src/Agent/NewRelic/Agent/Core/Segments/Segment.cs

```diff
@@ -449,6 +449,19 @@ public ISpan AddCustomAttribute(string key, object value)
             return this;
         }
 
+        public ISpan AddCloudSdkAttribute(string key, object value)
+        {
+            SpanAttributeValueCollection customAttribValues;
+            lock (_customAttribValuesSyncRoot)
+            {
+                customAttribValues = _customAttribValues ?? (_customAttribValues = new SpanAttributeValueCollection());
```

Comment:
Should the collection be renamed since it will no longer just contain the custom attribute values?

Reply:
Yes that's a good idea. If we decide to go this route I'll do a cleanup pass (I think this drops Code Coverage a little as well).